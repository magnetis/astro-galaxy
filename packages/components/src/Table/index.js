import PropTypes from 'prop-types';
import styled from 'styled-components';

const tableTheme = props =>
  props.darkMode
    ? {
        headColor: props.theme.colors.moon300,
        cellColor: props.theme.colors.moon100,
        rowColor: `${props.theme.colors.moon100}16`,
      }
    : {
        headColor: props.theme.colors.moon500,
        cellColor: props.theme.colors.moon900,
        rowColor: `${props.theme.colors.moon1000}08`,
      };

const Table = styled.table`
  border-collapse: collapse;
  width: 100%;

  td {
    color: ${props => tableTheme(props).cellColor};
    font: ${props => props.theme.fonts.secondary};
  }

  th {
    color: ${props => tableTheme(props).headColor};
    font: ${props => props.theme.fonts.primary};
  }

  td,
  th {
    padding: ${props => props.theme.space[4]}px ${props => props.theme.space[2]}px;
    text-align: inherit;

    &:first-child {
      padding-left: ${props => props.theme.space[4]}px;
    }

    &:last-child {
      padding-right: ${props => props.theme.space[4]}px;
    }
  }

  tr:nth-child(odd) td {
    background-color: ${props => tableTheme(props).rowColor};
  }

  tr[disabled] {
    opacity: 0.3;
  }
`;

Table.displayName = 'Table';

Table.propTypes = {
  darkMode: PropTypes.bool,
};

export default Table;
