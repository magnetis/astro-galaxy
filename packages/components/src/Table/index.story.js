import { theme } from '@magnetis/astro-galaxy-core';
import { IconPencil, IconTrash } from '@magnetis/astro-galaxy-icons';
import { boolean, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Table from '.';
import IconGhostButton from '../IconGhostButton';

const backgroundSelector = darkMode => ({
  backgrounds: [
    { name: 'space 100', value: theme.colors.space100, default: !darkMode },
    { name: 'moon 900', value: theme.colors.moon900, default: darkMode },
  ],
});

const DemoTable = props => (
  <Table darkMode={boolean('dark mode', props.darkMode)}>
    <thead>
      <tr>
        <th>Category</th>
        <th>Fund</th>
        <th>Investment</th>
        <th>&nbsp;</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Monthly Income</td>
        <td>FI Alaska Private Plus</td>
        <td>$ 87,900.00</td>
        <td>
          <IconGhostButton variant="ghost.uranus">
            <IconPencil />
          </IconGhostButton>
          <IconGhostButton variant="ghost.mars">
            <IconTrash />
          </IconGhostButton>
        </td>
      </tr>
      <tr>
        <td>Retirement Plan</td>
        <td>Icatu FAFI Future</td>
        <td>$ 143,273.00</td>
        <td>
          <IconGhostButton variant="ghost.uranus">
            <IconPencil />
          </IconGhostButton>
          <IconGhostButton variant="ghost.mars">
            <IconTrash />
          </IconGhostButton>
        </td>
      </tr>
      <tr disabled>
        <td>Stocks</td>
        <td>ETF Small 11</td>
        <td>$ 1,230.35</td>
        <td>
          <IconGhostButton variant="ghost.uranus">
            <IconPencil />
          </IconGhostButton>
          <IconGhostButton variant="ghost.mars">
            <IconTrash />
          </IconGhostButton>
        </td>
      </tr>
    </tbody>
  </Table>
);

storiesOf('tables|tables', module)
  .addDecorator(withKnobs)
  .add('default', () => <DemoTable />, backgroundSelector(false))
  .add('dark mode', () => <DemoTable darkMode />, backgroundSelector(true));
