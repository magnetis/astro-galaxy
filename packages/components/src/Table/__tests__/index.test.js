import React from 'react';
import Table from '..';

describe('Table', () => {
  it('should render correctly', () => {
    const component = rendererCreateWithTheme(<Table />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
