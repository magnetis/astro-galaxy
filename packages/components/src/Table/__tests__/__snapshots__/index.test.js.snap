// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render correctly 1`] = `
.c0 {
  border-collapse: collapse;
  width: 100%;
}

.c0 td {
  color: #32383c;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
}

.c0 th {
  color: #597183;
  font: normal normal 600 normal 16px/1.5 Poppins,sans-serif;
}

.c0 td,
.c0 th {
  padding: 32px 16px;
  text-align: inherit;
}

.c0 td:first-child,
.c0 th:first-child {
  padding-left: 32px;
}

.c0 td:last-child,
.c0 th:last-child {
  padding-right: 32px;
}

.c0 tr:nth-child(odd) td {
  background-color: #1c212408;
}

.c0 tr[disabled] {
  opacity: 0.3;
}

<table
  className="c0"
/>
`;
